src/prop/sat_proof_manager.h

```diff
@@ -40,6 +40,45 @@ namespace prop {
 
 class CnfStream;
 
+/**
+ * Works exactly as the regular buffered proof generator but when retrieving the
+ * proof of a step, we reuse the same ASSUMPTION step for repeated
+ * premises. This can be done because in the SAT proof there are no SCOPE steps,
+ * so there is no danger of mixing the scopes of assumptions.
+ */
+class SatBufferedProofGenerator : public BufferedProofGenerator
```

Comment:
Can we make this a flag on BufferedProofGenerator, like `mkUniqueAssume`? Similar to `autoSym` for CDProof.

Reply:
Done